### ISSUE TITLE
Failed Network Scan

### ISSUE THREAD

**Issue by fbbfnc** [anonymouz4/JamWiFi#1]
First of all thank you very much for you work. I'd like to help you if possible. 

Is there a way to enable debug mode in the XCode project? 
By the way, my issue is the following message when i press "Scan": A network scan could not be completed at this time.

**Comment by anonymouz4** [comment]
I've just tested it on 3 different devices and it works.

Make sure this file exists:
```/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Apple80211```

**Comment by anonymouz4** [comment]
If there‘s no further follow-up, I‘ll close this issue

**Comment by ajkblue** [comment]
I'm getting the same error, (on a 2019 MBP) I do have the Apple80211 file, and I have tried launching it via the Terminal with sudo su, as well as launching through Xcode as root.

All the console printed out was the following:
```
2020-03-27 11:07:51.107596-0600 JamWiFi[5075:82628] Main.m: Initializing..
JWDelegate: Launch Complete.
2020-03-27 11:07:51.347493-0600 JamWiFi[5075:82628] Metal API Validation Enabled
wifi scan error
wifi scan error
```

**Comment by anonymouz4** [comment]
@ajkblue It should now print a more detailed error

**Comment by ajkblue** [comment]
Alright, I just updated and now it's giving me this:
```
2020-03-27 23:48:48.677921-0600 JamWiFi[14268:346021] Main.m: Initializing..
JWDelegate: Launch Complete.
2020-03-27 23:48:48.887797-0600 JamWiFi[14268:346021] Metal API Validation Enabled
wifi scan error: Operation not supported on socket
```

**Comment by anonymouz4** [comment]
@ajkblue Does it never work or just sometimes?

**Comment by ajkblue** [comment]
It never works. All I can do is deauth from the network and then clicking on “Scan” causes the pop up “ A network scan could not be completed at this time.”

**Comment by anonymouz4** [comment]
@ajkblue Try replacing the following:

```
_ = _open!(&airportHandle)
_ = _bind!(airportHandle, self.interfaceName)
let result = _scan!(airportHandle, &foundNets, scanParams)
```
with smth like:
```
var test = _open!(&airportHandle)
print(test)
test= _bind!(airportHandle, self.interfaceName)
print(test)
let result = _scan!(airportHandle, &foundNets, scanParams)
```
and check if they both print 0

**Comment by ajkblue** [comment]
Yup, they both return a 0:
```
2020-03-28 20:09:19.931133-0600 JamWiFi[68755:1199358] Main.m: Initializing..
2020-03-28 20:09:20.097906-0600 JamWiFi[68755:1199358] Metal API Validation Enabled
JWDelegate: Launch Complete.
0
0
wifi scan error: Operation not supported on socket
```

**Comment by anonymouz4** [comment]
@ajkblue Then what preferences did you change/enable?

**Comment by ajkblue** [comment]
I don't think I've changed anything? This is what my preferences window looks like:
<img width="612" alt="Screen Shot 2020-03-29 at 10 36 17 AM" src="https://user-images.githubusercontent.com/9651939/77874013-dcec4d00-7208-11ea-8b87-4a7961d3fba6.png">

**Comment by ajkblue** [comment]
Is there anything else I can try? I've cleared out my Xcode cache and re-cloned the repo and went thought all of the steps listed above again, verified that the ``Apple80211`` file exists and I'm still getting the same error message. Even when changing the schemes in Xcode and launching as root. I also re-added the debug code which printed out 0 as shown below. I've verified that WiFi scanning and entering monitor mode works via Terminal.
![Sin título](https://user-images.githubusercontent.com/9651939/83334139-b165fd00-a261-11ea-864d-a453bc043bb0.png)
But I'm still getting the same error message in Xcode when launching the app, with or without running as root:

```
2020-05-30 10:35:09.399840-0600 JamWiFi[4313:37377] Main.m: Initializing..
2020-05-30 10:35:09.529450-0600 JamWiFi[4313:37377] Metal API Validation Enabled
JWDelegate: Launch Complete.
0
0
wifi scan error: Operation not supported on socket
0
0
wifi scan error: Operation not supported on socket
```
Now my JamWifi Preferences window looks like this:
![Sin título2](https://user-images.githubusercontent.com/9651939/83334306-c8f1b580-a262-11ea-9673-9fe91c856f5b.png)
